### PR TITLE
Update haproxy best practices using haproxy2.5.0

### PR DESCRIPTION
**Pull request by TomShawn**
### What is changed, added or deleted? (Required)
Update haproxy best practices using haproxy2.5.0

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/7046 https://github.com/pingcap/docs/pull/7082
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by TomShawn** [comment]
/cc @superlzs0476 @Chad20N13
PTAL. Thanks~

File: best-practices/haproxy-best-practices.md [hunk 1]
```diff
@@ -71,28 +71,51 @@ yum -y install epel-release gcc systemd-devel
...
 
+3. 从源码编译 HAProxy 应用：
+
+    {{< copyable "shell-regular" >}}
+
+    ```bash
+    cd haproxy-2.5.0
+    make clean
+    make -j 8 TARGET=linux-glibc USE_THREAD=1
+    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install  # 将 `${/app/haproxy}` 和 `${/app/haproxy/bin}` 替换为实际的路径。
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:102]:
```suggestion
    make PREFIX=${/app/haproxy} SBINDIR=${/app/haproxy/bin} install  # 将 `${/app/haproxy}` 和 `${/app/haproxy/bin}` 替换为自定义的实际路径。
```

File: best-practices/haproxy-best-practices.md [hunk 2]
```diff
@@ -145,10 +168,10 @@ global                                     # 全局配置。
    log         127.0.0.1 local2            # 定义全局的 syslog 服务器，最多可以定义两个。
    chroot      /var/lib/haproxy            # 更改当前目录并为启动进程设置超级用户权限，从而提高安全性。
    pidfile     /var/run/haproxy.pid        # 将 HAProxy 进程的 PID 写入 pidfile。
-   maxconn     4000                        # 每个 HAProxy 进程所接受的最大并发连接数。
+   maxconn     256                         # 每个 HAProxy 线程所接受的最大并发连接数。
```
Comment (lilin90) [inline best-practices/haproxy-best-practices.md:171]:
```suggestion
   maxconn     256                         # 每个 HAProxy 线程可接受的最大并发连接数。
```

**Review by lilin90 — APPROVED** [review]
LGTM

**Review by superlzs0476 — APPROVED** [review]
LGTM

**Pull request merged**